src/checkoutservice/main.go

```diff
@@ -130,6 +156,26 @@ func initStackDriverTracing() {
 	log.Warn("could not initialize stackdriver exporter after retrying, giving up")
 }
 
+// Initialize OTel trace provider that exports to Cloud Trace
+func initTraceProvider() {
+	projectID := os.Getenv("GOOGLE_CLOUD_PROJECT")
+	exporter, err := texporter.NewExporter(texporter.WithProjectID(projectID))
+	if err != nil {
+		log.Fatalf("failed to initialize exporter: %v", err)
```

Comment:
Not sure if we should exit the app when trace exporter fails to initialize. It makes our app more fragile and dependent on exporter implementation

Reply:
How the auth to Cloud Trace is handled? I assume it's likely using environment variable that is supposed to be set (GOOGLE_APPLICATION_CREDENTIALS). So worth ensuring this variable is set correctly

Reply:
yep it pulls from environment variables for auth
if running on gcp, I believe these are set automatically. otherwise it needs to be set manually or be provided a credentials file

Reply:
I'm having it retry up to 3 times to be a little less brittle. Info if it fails once, Warn if it fails three times.